tools/mtouch/Application.cs

```diff
@@ -1147,7 +1147,20 @@ void BuildFinalExecutable ()
 			} else {
 				if (!IsWatchExtension) {
 					// In extensions we need to save a list of the frameworks we need so that the main app can get them.
-					var all_frameworks = Frameworks.Union (WeakFrameworks);
+					var all_frameworks = new HashSet<string> ();
+					all_frameworks.UnionWith (Frameworks);
```

Comment:
I don't think I understand what you mean.

The `Union` method is an extension method on IEnumerable, which means that if I keep using it, I'll end up creating a lot of enumerables:

    var all_frameworks = Frameworks.Union (WeakFrameworks).
    foreach (var t in Targets) {
        all_frameworks = all_frameworks.Union (t.Frameworks);
        all_frameworks = all_frameworks.Union (t.WeakFrameworks);
        foreach (var a in t.Assemblies) {
            if (a.Frameworks != null)
                all_frameworks = all_frameworks.Union (a.Frameworks);
            if (a.WeakFrameworks != null)
                all_frameworks = all_frameworks.Union (a.WeakFrameworks);
        }
    }

is that what you meant?